profiles/factories.py

```diff
@@ -44,16 +45,29 @@ class LegalAddressFactory(DjangoModelFactory):
     """Factory for LegalAddress"""
 
     user = SubFactory("profiles.factories.UserFactory", legal_address=None)
-
     first_name = Faker("first_name")
     last_name = Faker("last_name")
-
     street_address_1 = Faker("street_address")
-
-    state_or_territory = Faker("lexify", text="??-??", letters=string.ascii_uppercase)
     city = Faker("city")
-    country = Faker("country_code", representation="alpha-2")
     postal_code = Faker("postalcode")
+    country = Faker("country_code", representation="alpha-2")
+
+    @lazy_attribute
+    def state_or_territory(self):
+        """
+        Return a state/territory if one is applicable for the given country.
+
+        NOTE: There didn't seem to be a straightforward way to produce a state/territory using faker,
+        and using random letters for that value results in addresses that fail our own address validation.
+        This implementation is here to ensure that we produce legit-looking addresses that our own app
+        will consider valid.
+        """
+        if self.country not in {"US", "CA"}:
+            return ""
+        subdivisions = pycountry.subdivisions.get(country_code=self.country)
+        subdivision = random.sample(subdivisions, 1)[0]
```

Comment:
Can we use Faker's random here so we can use the same seed value? https://faker.readthedocs.io/en/master/#accessing-the-random-instance